xarray/coding/cftimeindex.py

```diff
@@ -528,6 +528,83 @@ def strftime(self, date_format):
         """
         return pd.Index([date.strftime(date_format) for date in self._data])
 
+    @property
+    def asi8(self):
```

Comment:
This does seem useful. Shall we list it as public API?

Reply:
Yeah, that sounds like a good idea.  It looks like it is [already listed in the docs for `CFTimeIndex`](http://xarray.pydata.org/en/stable/generated/xarray.CFTimeIndex.html) (despite the fact that it currently returns `None`); is there anything else that I would need to do?  

I added a couple tests for good measure (including one that turned up a floating point issue that I fixed).